curriculum/challenges/english/25-front-end-development/quiz-javascript-higher-order-functions/66edcdd18a4ef8df16e6bb7e.md

```diff
@@ -17,439 +17,482 @@ Answer all of the questions below correctly to pass the quiz.
 
 #### --text--
 
-Placeholder question
+Consider the following statements about JavaScript higher-order functions. Which one presents a common misconception?
 
 #### --distractors--
 
-Placeholder distractor 1
+Higher-order functions can greatly enhance code readability and maintainability by enabling functional programming techniques.
 
 ---
 
-Placeholder distractor 2
+Higher-order functions like map, filter, and reduce are powerful tools for array manipulation, but they are not unique to functional programming.
 
 ---
 
-Placeholder distractor 3
+Higher-order functions may introduce complexity in understanding code, but they can also lead to more expressive and concise solutions.
 
 #### --answer--
 
-Placeholder answer
+All functions in JavaScript, including those that do not manipulate or return other functions, can be classified as higher-order functions.Placeholder answer
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following statements about closures and higher-order functions is true?
 
 #### --distractors--
 
-Placeholder distractor 1
+Higher-order functions must always return a new function; otherwise, they are not considered higher-order.
 
 ---
 
-Placeholder distractor 2
+Closures and higher-order functions are unrelated concepts in JavaScript.
 
 ---
 
-Placeholder distractor 3
+Higher-order functions cannot create closures since they do not return inner functions.
 
 #### --answer--
 
-Placeholder answer
+Closures created by higher-order functions can capture the lexical scope, allowing for state preservation.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Evaluate the Output
+
+```js
+const numbers = [1, 1, 1, 1, 1];
+let sum = 0;
+const forEachRes = numbers.forEach(num => {
+  return (sum += num);
+});
+const mapRes = numbers.map(num => {
+  return (sum += num);
+});
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+forEachRes is undefined and mapRes is [1,2,3,4,5]
 
 ---
 
-Placeholder distractor 2
+forEachRes is 0 and mapRes is [1,2,3,4,5]
 
 ---
 
-Placeholder distractor 3
+forEachRes is 5 and mapRes is [1,2,3,4,5]
 
 #### --answer--
 
-Placeholder answer
+forEachRes is undefined and mapRes is [6,7,8,9,10]
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Evaluate the output
+
+```js
+[, undefined, 'a', 'b', { 20: 5 }].sort();
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+unsupported elements for an array to be sorted, hence error
 
 ---
 
-Placeholder distractor 2
+Callback not supplied, hence error
 
 ---
 
-Placeholder distractor 3
+```js
+[<1 empty item> , 'a', 'b', undefined,{ '20': 5 } ]
+```
 
 #### --answer--
 
-Placeholder answer
+```js
+[ { '20': 5 }, 'a', 'b', undefined, <1 empty item> ]
+```
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following describes a callback function in JavaScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+## A function that is called immediately upon declaration.
```

Comment:
why do you have this as an header?

```suggestion
A function that is called immediately upon declaration.
```

Reply:
Hey,
I did not notice that mustve been a prettier autoformatting
Will fix it